src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,251 @@
+        File f = new File(filePath);
+        try {
+            if(f.createNewFile()) {
+                System.out.println("New file is created in the folder for storing task list.");
+            }
+        } catch(IOException e) {
+            System.out.println("An error has occurred.");
+        }
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String thisLine = s.nextLine();
+            String[] words = thisLine.split("\\|", 4);
+            switch(words[0].trim()) {
+            case "T":
+                tasks.add(new Todo(words[2].trim()));
+                break;
+            case "E":
+                tasks.add(new Event(words[2].trim(), words[3].trim()));
+                break;
+            case "D":
+                tasks.add(new Deadline(words[2].trim(), words[3].trim()));
+                break;
+            default:
+                System.out.println("This line is not added to list:" + thisLine);
+            }
+            if(Integer.parseInt(words[1].trim())==1) {
+                tasks.get(totalTasks).markAsDone();
+            }
+            totalTasks++;
+        }
+    }
+
+    private static void saveTaskList() throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for(int i = 0; i< totalTasks; ++i) {
+            fw.write(tasks.get(i).storeString() + "\n");
+        }
+        fw.close();
+    }
+
+    private static void addTodo(String task) {
+        tasks.add(new Todo(task));
+    }
+
+    private static void addEvent(String task) {
+        String[] words = task.split("/at");
+        tasks.add(new Event(words[0], words[1]));
+    }
+
+    private static void addDeadline(String task) {
+        String[] words = task.split("/by");
+        tasks.add(new Deadline(words[0], words[1]));
+    }
+
+    private static void updateDone(int updateNumber) {
+        tasks.get(updateNumber).markAsDone();
+    }
+
+    private static void updateDelete(int deleteNumber) {
+        tasks.remove(deleteNumber);
+    }
+
+    private static void printList() {
+        if(totalTasks==0) {
+            System.out.println("List is empty.");
+            return;
+        }
+        System.out.println("Here are the tasks in your list:");
+        for(int taskNumber = 0; taskNumber < totalTasks; taskNumber++) {
+            System.out.println(taskNumber + 1 + "." + tasks.get(taskNumber));
+        }
+    }
+
+    private static void printDone(int updateNumber) {
+        System.out.println("Nice! I've marked this task as done: ");
+        System.out.println(tasks.get(updateNumber));
+    }
+
+    private static void printDelete(int deleteNumber) {
+        System.out.println("Noted. I've removed this task: ");
+        System.out.println(tasks.get(deleteNumber));
+        totalTasks -= 1;
+        System.out.println((totalTasks==1)? "Now you have 1 task in the list." : "Now you have " + totalTasks + " tasks in the list.");
```

Comment:
I think you exceeded the line length of more than `120` characters here. You can try to add a line break to break this into a 2-line code instead.